VRDR.Messaging/BaseMessage.cs

```diff
@@ -261,30 +261,44 @@ public string MessageSource
         }
 
         /// <summary>Message Destination</summary>
-        /// <value>the message destinations, in csv format to support multiple endpoints while maintaining backwards compatability.</value>
+        /// <value>the message destinations, in csv format to support multiple endpoints. Acts as a wrapper for MessageDestinations while still maintaining backwards compatibility.</value>
         public string MessageDestination
         {
             get
             {
-                if (Header?.Destination == null || Header.Destination.Count() <= 1) {
-                    return Header?.Destination?.FirstOrDefault()?.Endpoint;
+                List<string> destinations = this.MessageDestinations;
+                if (destinations == null || (destinations.Count() == 1 && destinations[0] == null) || destinations.Count() < 1) {
+                    return null;
                 }
-                return String.Join(",", Header?.Destination?.Select(dest => dest.Endpoint));
+                return String.Join(",", this.MessageDestinations);
+            }
+            set
+            {
+                this.MessageDestinations = value != null ? value.Split(',').ToList() : null;
+            }
+        }
+
+        /// <summary>Message Destinations</summary>
+        /// <value>the message destinations in list-based format.</value>
+        public List<string> MessageDestinations
```

Comment:
Maybe should be called "MessageDestinationList" or "MessageDestinationEndpoints" or something to avoid confusion?

Reply:
I recommend sticking with the name MessageDestinations: it's unambiguous in that it's plural and the message signature makes it clear what's expected. It also aligns better with other places in the code where we expect a list, e.g.

```
public List<Issue> Issues
```

from ErrorMessage.cs in the Messaging library.

Reply:
Good point.